src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.SolutionChecksumUpdater.cs

```diff
@@ -112,16 +119,23 @@ public override void Shutdown()
 
             private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
             {
-                // special initial case
-                if (e.Kind == WorkspaceChangeKind.SolutionAdded)
+                // special bulk update case
+                if (e.Kind == WorkspaceChangeKind.SolutionAdded ||
+                    e.Kind == WorkspaceChangeKind.ProjectAdded)
                 {
-                    CreateInitialSolutionChecksum();
+                    _synchronize = true;
+                    EnqueueChecksumUpdate();
```

Comment:
I intentionally didn't do that so when there is bulk change, it doesnt get delayed.